common/utils/ti_fec_decode.cpp

```diff
@@ -0,0 +1,313 @@
+    // used)
+    nMinCost = 0xFF;
+    // Get 2b input symbol (MSB first) and do one iteration of Viterbi decoding
+    if ((iBit -= 2) < 0) {
+      iBit = 6;
+      pInData++;  // Update pointer to the next byte of received data
+    }
+
+    // For each destination state in the trellis, calculate hamming costs for
+    // both possible paths into state and select the one with lowest cost.
+
+#if 1
+// Unrolling this loop seems to net us 5-7% speed improvement
+#define DO_STATE(iDestState)                                                  \
+  {                                                                           \
+    nInputBit = (iDestState % 2);                                             \
+    /* Calculate cost of transition from each of the two source states */     \
+    /* (cost is Hamming difference between received 2b symbol and expected */ \
+    /* symbol for transition) */                                              \
+    iSrcState0 = aTrellisSourceStateLut[iDestState][0];                       \
+    iSrcState1 = aTrellisSourceStateLut[iDestState][1];                       \
+    nCost0 = nCost[iLastBuf][iSrcState0];                                     \
+    nCost1 = nCost[iLastBuf][iSrcState1];                                     \
+    /* Being smart and writing down the index once nets us another 7.5% */    \
+    /* But being cleaver and removing this increment doesn't change it */     \
+    index = destStateLUTIndex(iDestState, symbol, 0);                         \
+    nCost0 += costStepLUT[index];                                             \
+    nCost1 += costStepLUT[index + 1];                                         \
+    /* Select transition that gives lowest cost in destination state, copy */ \
+    /* that source state's path and add new decoded bit*/                     \
+    if (nCost0 <= nCost1) {                                                   \
+      nCost[iCurrBuf][iDestState] = nCost0;                                   \
+      nMinCost = MIN(nMinCost, nCost0);                                       \
+      aPath[iCurrBuf][iDestState] =                                           \
+          (aPath[iLastBuf][iSrcState0] << 1) | nInputBit;                     \
+    } else {                                                                  \
+      nCost[iCurrBuf][iDestState] = nCost1;                                   \
+      nMinCost = MIN(nMinCost, nCost1);                                       \
+      aPath[iCurrBuf][iDestState] =                                           \
+          (aPath[iLastBuf][iSrcState1] << 1) | nInputBit;                     \
+    }                                                                         \
+  }
+
+    DO_STATE(0);
+    DO_STATE(1);
+    DO_STATE(2);
+    DO_STATE(3);
+    DO_STATE(4);
+    DO_STATE(5);
+    DO_STATE(6);
+    DO_STATE(7);
+
+#else
+
+    // might be a minimal speedup but u could cut down on all the calls to
```

Comment:
If you think this might get used again in the future I'm fine with it staying here commented out, otherwise probably better to remove it.